src/metabase-report-notifier/src/main/java/com/champtitles/metabasereportexecutor/notifier/EmailSender.java

```diff
@@ -0,0 +1,216 @@
+package com.champtitles.metabasereportexecutor.notifier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.activation.DataHandler;
+import javax.activation.DataSource;
+import javax.mail.*;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+import javax.mail.util.ByteArrayDataSource;
+import java.util.Properties;
+
+public class EmailSender {
+
+    private static final Logger logger = LoggerFactory.getLogger(EmailSender.class.getName());
+    private final Session session;
+    private final String fromAddress;
+    private final EmailTransport emailTransport;
+
+    /**
+     * @param smtpHost     host of SMTP server
+     * @param smtpPort     port of SMTP server
+     * @param smtpUser     username to log in to SMTP server
+     * @param smtpPassword password to log in to SMTP server
+     * @param fromAddress  email address to use as the sender
+     */
+    public EmailSender(String smtpHost, String smtpPort, String smtpUser, String smtpPassword, String fromAddress) {
+        this(smtpHost, smtpPort, smtpUser, smtpPassword, fromAddress, new EmailTransport());
+    }
+
+    EmailSender(String smtpHost, String smtpPort, String smtpUser, String smtpPassword, String fromAddress, EmailTransport emailTransport) {
+        Properties smtpProperties = createSmtpProperties(smtpHost, smtpPort, smtpUser, smtpPassword);
+        Authenticator smtpAuthenticator = createSmtpAuthenticator(smtpUser, smtpPassword);
+        session = Session.getInstance(smtpProperties, smtpAuthenticator);
+        this.fromAddress = fromAddress;
+        this.emailTransport = emailTransport;
+    }
+
+    /**
+     * Create an email with a xlsx attachment and send it
+     *
+     * @param subject    subject line of the email
+     * @param recipients list of recipient addresses
+     * @param fileName   name of the xlsx file attachment
+     * @param xlsxData   contents of the xlsx file attachment
+     */
+    public void sendEmail(String subject, String[] recipients, String fileName, byte[] xlsxData) {
+        logger.info("creating email message");
```

Comment:
Remove log statements without value (or add information to this so it has value)